webui/react/src/components/Badge.tsx

```diff
@@ -10,12 +10,14 @@ import { ResourceState, RunState, SlotState } from 'types';
 
 import css from './Badge.module.scss';
 
-export enum BadgeType {
-  Default = 'Default',
-  Header = 'Header',
-  Id = 'Id',
-  State = 'State',
-}
+export const BadgeType = {
+  Default: 'Default',
+  Header: 'Header',
+  Id: 'Id',
+  State: 'State',
+} as const;
+
+export type BadgeType = typeof BadgeType[keyof typeof BadgeType];
```

Comment:
Can we add a `ValueOf` utility type since we're now using this pattern so often? It just makes the code easier to read.

Reply:
make sense